pyomo/contrib/pyros/config.py

```diff
@@ -0,0 +1,1043 @@
+"""
+Interfaces for managing PyROS solver options.
+"""
+
+from collections.abc import Iterable
+import logging
+import os
+
+from pyomo.common.collections import ComponentSet
+from pyomo.common.config import (
+    ConfigDict,
+    ConfigValue,
+    In,
+    NonNegativeFloat,
+    InEnum,
+    Path,
+)
+from pyomo.common.errors import ApplicationError
+from pyomo.core.base import Var, _VarData
+from pyomo.core.base.param import Param, _ParamData
+from pyomo.opt import SolverFactory
+from pyomo.contrib.pyros.util import ObjectiveType, setup_pyros_logger
+from pyomo.contrib.pyros.uncertainty_sets import UncertaintySetDomain
+
+
+default_pyros_solver_logger = setup_pyros_logger()
+
+
+class LoggerType:
+    """
+    Domain validator for objects castable to logging.Logger.
+    """
+
+    def __call__(self, obj):
+        """
+        Cast object to logger.
+
+        Parameters
+        ----------
+        obj : object
+            Object to be cast.
+
+        Returns
+        -------
+        logging.Logger
+            If `str_or_logger` is of type `logging.Logger`,then
+            `str_or_logger` is returned.
+            Otherwise, ``logging.getLogger(str_or_logger)``
+            is returned.
+        """
+        if isinstance(obj, logging.Logger):
+            return obj
+        else:
+            return logging.getLogger(obj)
+
+    def domain_name(self):
+        """Return str briefly describing domain encompassed by self."""
+        return "None, str or logging.Logger"
+
+
+class PositiveIntOrMinusOne:
+    """
+    Domain validator for objects castable to a
+    strictly positive int or -1.
+    """
+
+    def __call__(self, obj):
+        """
+        Cast object to positive int or -1.
+
+        Parameters
+        ----------
+        obj : object
+            Object of interest.
+
+        Returns
+        -------
+        int
+            Positive int, or -1.
+
+        Raises
+        ------
+        ValueError
+            If object not castable to positive int, or -1.
+        """
+        ans = int(obj)
+        if ans != float(obj) or (ans <= 0 and ans != -1):
+            raise ValueError(f"Expected positive int or -1, but received value {obj!r}")
+        return ans
+
+    def domain_name(self):
+        """Return str briefly describing domain encompassed by self."""
+        return "positive int or -1"
+
+
+class PathLikeOrNone:
```

Comment:
I don't think this needs a separate validator.  We should incorporate the use of `fsdecode` into the `pyomo.common.config.Path` validator.  You also don't need special handling for `None`: All `ConfigValue` instances admit `None` as a valid value.